File: Fundamentals_of_Software_Design/1_Basic_skills/Очистка строки/main.py
def strclear(string):
    i = 0
    result = []
    space_found = False  # Флаг, чтобы отслеживать, если был пробел подряд

    # Перебираем строку
    while i < len(string):
        if string[i].isspace():
            if not space_found:  # Если пробела еще не было, то ставим один
                result.append(' ')
                space_found = True
        else:
            result.append(string[i])  # Просто добавляем символ
            space_found = False  # Сброс флага
        i += 1

    return ''.join(result)

# Пример использования
input_str = "ab    cd"
print(f"<{input_str}>")  # <ab    cd>
output_str = strclear(input_str)
print(f"<{output_str}>")  # <ab cd>

File: Fundamentals_of_Software_Design/1_Basic_skills/Очистка строки/main.c
#include <stdio.h>
#include <ctype.h>

void strclear(char *string) {
    int i = 0, j = 0;
    int space_found = 0;  // Флаг, чтобы отслеживать, если был пробел подряд

    // Перебираем строку
    while (string[i] != '\0') {
        if (isspace(string[i])) {
            if (!space_found) {  // Если пробела еще не было, то ставим один
                string[j++] = ' ';
                space_found = 1;
            }
        } else {
            string[j++] = string[i];  // Просто копируем символ
            space_found = 0;  // Сброс флага
        }
        i++;
    }

    string[j] = '\0';  // Завершаем строку
}

int main() {
    char buf[] = "ab    cd";
    printf("<%s>\n", buf);  // <ab    cd>
    strclear(buf);
    printf("<%s>\n", buf);  // <ab cd>
    return 0;
}
